src/phoenix/datasets/evaluators/utils.py

```diff
@@ -0,0 +1,164 @@
+import functools
+import inspect
+from typing import Any, Callable, Optional
+
+from phoenix.datasets.types import (
+    AnnotatorKind,
+    CanAsyncEvaluate,
+    CanEvaluate,
+    EvaluationResult,
+    EvaluatorCallable,
+    Example,
+    ExperimentEvaluator,
+    ExperimentRun,
+    JSONSerializable,
+)
+
+
+def _unwrap_json(obj: JSONSerializable) -> JSONSerializable:
+    if isinstance(obj, dict):
+        if len(obj) == 1:
+            key = next(iter(obj.keys()))
+            output = obj[key]
+            assert isinstance(
+                output, (dict, list, str, int, float, bool, type(None))
+            ), "Output must be JSON serializable"
+            return output
+    return obj
+
+
+def _validate_signature(sig: inspect.Signature) -> None:
+    # Check that the wrapped function has a valid signature for use as an evaluator
+    # If it does not, raise an error to exit early before running evaluations
+    params = sig.parameters
+    valid_named_params = {"experiment_input", "output", "reference", "metadata"}
+    if len(params) == 0:
+        raise ValueError("Evaluation function must have at least one parameter.")
+    if len(params) > 1:
+        not_found = set(params) - valid_named_params
+        if not_found:
+            raise ValueError(
+                (
+                    f"Invalid parameter names in evaluation function: {', '.join(not_found)}. "
+                    "Parameters names for multi-argument functions must be "
+                    f"any of: {', '.join(valid_named_params)}."
+                )
+            )
+
+
+def _bind_signature(
+    sig: inspect.Signature, example: Example, experiment_run: ExperimentRun
+) -> inspect.BoundArguments:
+    if experiment_run.output:
+        output = experiment_run.output.result
+    else:
+        output = None
+    parameter_mapping = {
+        "experiment_input": example.input,
```

Comment:
why would we not just map input?

Reply:
`input` is a reserved python keyword, I think it's preferable not to clobber it

Reply:
Really? That seems crazy...

Reply:
because `input` is used inside the local function scope I think it isn't so bad if we switch this to input, my questions are: is this good form & will people want to do that? maybe let's see what everyone else thinks?

Reply:
I think people will want it to be input and it's function scope so it's a non issue. I don't want to type `example_input`